src/massive/munit/Assert.hx

```diff
@@ -226,4 +226,25 @@ class Assert
 	{
 		throw new AssertionException(msg, info);
 	}
+	
+	/**
+	 * Assert that an expectation was thrown. Can expect strings and non-strings.
+	 *
+	 * @param   code				a function which should throw an exception 
+	 * @return					  the exception that was thrown	  
+	 * @throws  AssertionException  if no expectation is thrown
+	 */
+	public static function throws(code:Dynamic, ?info:PosInfos):Dynamic
+	{
+		try
+		{
+			code();
+			fail("Expected exception wasn't thrown!", info);
+			return null; // needed to compile
+		}
+		catch (e:Dynamic)
```

Comment:
I think it's better to have a simpler `throws` method. The user just has to narrow down which line of code should throw. If they care about the type, they can write their own if/else code (like you've described) to test the type more specifically.

Plus, with string errors, the code looks a little uglier:

```
var message:String = Assert.throws(String, function() { ... })
Assert.areEqual("Invalid data", message);
```

Instead of this:

```
var message:String = Assert.throws(function() { ... })
Assert.areEqual("Invalid data", message);
```

I don't see a big difference in terms of code clarity between these two options, either.

If you insist, let me know, and I'll change the method to what you've described.